.github/workflows/update-changelog.yml

```diff
@@ -2,7 +2,7 @@ name: "Update Changelog"
 
 on:
   release:
-    types: [published]
```

Comment:
here is relevant bit of documentation: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

makes sense.